### PR TITLE
rules*: use tokenAssert methods

### PR DESCRIPTION
**Pull request by hzoo**
- requireCurlyBraces: use tokenAssert (make it similar to disallowCurlyBraces)

### CODE REVIEW

File: lib/rules/require-curly-braces.js [hunk 1]
```diff
@@ -97,14 +91,12 @@ module.exports.prototype = {
...
+                if (typeIndex['if'] && isNotABlockStatement(node.consequent)) {
                     addError('If', node);
                 }
-                if (typeIndex['else'] && node.alternate &&
-                    node.alternate.type !== 'BlockStatement' &&
-                    node.alternate.type !== 'IfStatement'
-                ) {
-                    addError('Else', node);
+                if (typeIndex['else'] && isNotABlockStatement(node.alternate) &&
+                    node.alternate.type !== 'IfStatement') {
```
Comment (hzoo) [inline lib/rules/require-curly-braces.js]: ` && node.alternate.type !== 'IfStatement'` is kept b/c otherwise some of the tests regarding presets failed

```
modules/string-checker presets preset crockford should not report any errors from the sam
ple file:
     AssertionError: false == true
```

Not sure if it means we need to add that to disallow

Comment (mikesherov) [inline lib/rules/require-curly-braces.js]: Yes, it needs to be kept only for the require case, because otherwise `else if...` would need to be `else { if...` because technically that else doesn't have curly braces.

Comment (mikesherov) [inline lib/rules/require-curly-braces.js]: But that also means we should add a test for `else if` to this PR in the require-curly-braces suite so we don't lose coverage should we ever remove the preset tests.

Comment (hzoo) [inline lib/rules/require-curly-braces.js]: Oh ok that makes sense

**Comment by hzoo** [comment]
added some other files that I missed before

File: lib/rules/disallow-space-between-arguments.js [hunk 2]
```diff
@@ -46,20 +46,13 @@ module.exports.prototype = {
...
-
-        var tokens = file.getTokens();
-
         file.iterateNodesByType(['CallExpression'], function(node) {
-
             node.arguments.forEach(function(param, i) {
-
-                var punctuatorToken = file.getTokenByRangeStart(param.range[1]);
-
+                var punctuatorToken = file.getPrevToken(file.getFirstNodeToken(param));
```
Comment (zxqfox) [inline lib/rules/disallow-space-between-arguments.js]: You using here `punctuatorToken = getPrevToken` and later `nextToken: file.getNextToken(punctuatorToken)`.

So `file.getFirstNodeToken` is already a `nextToken` for `punctuatorToken`.

You can get rid of unnecessary `getNetToken` call by storing `firstNodeToken`.

File: lib/rules/require-spaces-in-for-statement.js [hunk 3]
```diff
@@ -58,22 +58,19 @@ module.exports.prototype = {
...
     check: function(file, errors) {
         file.iterateNodesByType('ForStatement', function(node) {
-            var prevToken;
             if (node.test) {
-                var testToken = file.getTokenByRangeStart(node.test.range[0]);
-                prevToken = file.getPrevToken(testToken);
+                var testToken = file.getFirstNodeToken(node.test);
                 errors.assert.whitespaceBetween({
-                    token: prevToken,
+                    token: file.getPrevToken(testToken),
```
Comment (zxqfox) [inline lib/rules/require-spaces-in-for-statement.js]: What if there is a block comment between these tokens?

File: lib/rules/disallow-space-between-arguments.js (hunk 2 above)
Comment (hzoo) [inline lib/rules/disallow-space-between-arguments.js]: The `punctuatorToken.value === ','` is important
Ok I should just rename to this then.

```js
var nextToken = file.getFirstNodeToken(param);
                var punctuatorToken = file.getPrevToken(nextToken);
                if (punctuatorToken.value === ',') {
                    errors.assert.noWhitespaceBetween({
                        token: punctuatorToken,
                        nextToken: file.getNextToken(punctuatorToken)
                    });
                }
```

File: lib/rules/require-spaces-in-for-statement.js (hunk 3 above)
Comment (hzoo) [inline lib/rules/require-spaces-in-for-statement.js]: How are we solving this for all other rules - maybe that should be part of the whitespaceBetween method? I don't know how common it is to have comments in for loop statements

Comment (mikesherov) [inline lib/rules/require-spaces-in-for-statement.js]: Right, only certain rules where it's common to have comment blocks in between (such as function arguments) should we consider them. But again, I think it's beyond scope of this pull request. @zxqfox, file a new issue for that please.

File: lib/rules/disallow-space-between-arguments.js [hunk 4]
```diff
@@ -46,20 +46,13 @@ module.exports.prototype = {
...
                 if (punctuatorToken.value === ',') {
                     errors.assert.noWhitespaceBetween({
                         token: punctuatorToken,
-                        nextToken: file.getNextToken(punctuatorToken),
-                        message: 'Illegal space between arguments'
```
Comment (mikesherov) [inline lib/rules/disallow-space-between-arguments.js]: why is this removed?

File: lib/rules/disallow-space-between-arguments.js (hunk 2 above)
Comment (mikesherov) [inline lib/rules/disallow-space-between-arguments.js]: why isn't this line just: `var punctuatorToken = file.getFirstNodeToken(param);` previously it was `var punctuatorToken = file.getTokenByRangeStart(param.range[1]);`?

Comment (mikesherov) [inline lib/rules/disallow-space-between-arguments.js]: never mind, I get it.

File: test/rules/require-curly-braces.js [hunk 5]
```diff
@@ -153,4 +153,14 @@ describe('rules/require-curly-braces', function() {
         assert(error.line === 4);
         assert(error.column === 12);
     });
+
+    it('should not report missing `else` braces for `else if`', function() {
+        checker.configure({ requireCurlyBraces: ['else'] });
+        assert(checker.checkString('if (x) { x++; } else if (x) x++;').isEmpty());
```
Comment (mikesherov) [inline test/rules/require-curly-braces.js]: please change this to: `assert(checker.checkString('if (x) { x++; } else if (x) { x++; }').isEmpty());` it is misleading otherwise.

File: test/rules/require-curly-braces.js [hunk 6]
```diff
@@ -153,4 +153,14 @@ describe('rules/require-curly-braces', function() {
...
+    });
+
+    it('should report missing `if` braces for `else if`', function() {
```
Comment (mikesherov) [inline test/rules/require-curly-braces.js]: remove this test, it's not needed.

**Comment by mikesherov** [comment]
Thanks again @hzoo for all your hard work here.

File: lib/rules/disallow-space-between-arguments.js (hunk 2 above)
Comment (zxqfox) [inline lib/rules/disallow-space-between-arguments.js]: :+1:

File: lib/rules/require-spaces-in-for-statement.js (hunk 3 above)
Comment (zxqfox) [inline lib/rules/require-spaces-in-for-statement.js]: @hzoo I've thought about it, but better no ;-) We did this: [getCommentAfterToken](https://github.com/jscs-dev/node-jscs/search?utf8=%E2%9C%93&q=getCommentAfterToken&type=Code) — helpers to get all comments before and after token.

→ #987

File: lib/rules/disallow-space-between-arguments.js (hunk 4 above)
Comment (hzoo) [inline lib/rules/disallow-space-between-arguments.js]: Wasn't sure if we wanted to just use the default messages or not

**Comment by hzoo** [comment]
Made the suggested changes

**Comment by mikesherov** [comment]
@hzoo, please rebase. I've already changed the callExpressions rules as part of a code coverage sweep.

**Comment by hzoo** [comment]
Ok rebased and changed the last commit (requireSpaces) to match the disallow one

**Comment by mikesherov** [comment]
Thanks again!
